packages/airnode-protocol/contracts/utils/SelfServeRrpBeaconServerWhitelister.sol

```diff
@@ -0,0 +1,126 @@
+        for (
+            uint256 ind = 0;
+            ind < _beaconIdsExpirationTimestamp.length;
+            ind++
+        ) {
+            beaconIdToExpirationTimestamp[
+                _beaconIdsExpirationTimestamp[ind]
+            ] = _expirationTimestamps[ind];
+        }
+        for (
+            uint256 ind = 0;
+            ind < _beaconIdsIndefiniteWhitelistStatus.length;
+            ind++
+        ) {
+            beaconIdToIndefiniteWhitelistStatus[
+                _beaconIdsIndefiniteWhitelistStatus[ind]
+            ] = _indefiniteWhitelistStatuses[ind];
+        }
+    }
+
+    /// @notice Adds a new beaconId with an expiration timestamp
+    /// that can be whitelisted by readers
+    /// @param _beaconId The beaconId to set an expiration timestamp for
+    /// @param _expirationTimestamp The expiration timestamp for the beaconId
+    function setBeaconIdWithExpirationTimestamp(
+        bytes32 _beaconId,
+        uint64 _expirationTimestamp
+    ) external override onlyOwner {
+        beaconIdToExpirationTimestamp[_beaconId] = _expirationTimestamp;
+        emit SetBeaconIdWithExpirationTimestamp(
+            _beaconId,
+            _expirationTimestamp
+        );
+    }
+
+    /// @notice Adds a new beaconId with an indefinite whitelist status
+    /// that can be whitelisted by readers indefinetly
+    /// @param _beaconId The beaconId to set an indefinite whitelist status for
+    /// @param _indefiniteWhitelistStatus The indefinite whitelist status for the beaconId
+    function setBeaconIdWithIndefiniteWhitelistStatus(
+        bytes32 _beaconId,
+        bool _indefiniteWhitelistStatus
+    ) external override onlyOwner {
+        beaconIdToIndefiniteWhitelistStatus[
+            _beaconId
+        ] = _indefiniteWhitelistStatus;
+        emit SetBeaconIdWithIndefiniteWhitelistStatus(
+            _beaconId,
+            _indefiniteWhitelistStatus
+        );
+    }
+
+    /// @notice Whitelists a reader on the RrpBeaconServer with an expiration timestamp
+    /// @param _beaconId The beaconId to whitelist
+    /// @param _reader The reader to whitelist on the beaconId
+    function whitelistReaderWithExpiration(bytes32 _beaconId, address _reader)
+        external
+        override
+    {
+        uint64 expirationTimestamp = beaconIdToExpirationTimestamp[_beaconId];
+        require(
+            expirationTimestamp > block.timestamp,
+            "Cannot whitelist for beacon"
+        );
+        IRrpBeaconServer(rrpBeaconServer).setWhitelistExpiration(
+            _beaconId,
+            _reader,
+            expirationTimestamp
+        );
+        emit WhitelistedReaderWithExpiration(_beaconId, _reader);
+    }
+
+    /// @notice Whitelists a reader on the RrpBeaconServer with an indefinite whitelist status
+    /// @param _beaconId The beaconId to whitelist
+    /// @param _reader The reader to whitelist on the beaconId
+    function whitelistReaderIndefinitely(bytes32 _beaconId, address _reader)
```

Comment:
So we can't really set the indefinite whitelist status back to `false` using this contract, correct? Wouldn't it be better to have a setter redirect instead?